library/management/chat/src/main/java/org/quiltmc/qsl/chat/api/types/AbstractChatMessage.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2023 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.chat.api.types;
+
+import net.minecraft.entity.player.PlayerEntity;
+import org.jetbrains.annotations.NotNull;
+import org.quiltmc.qsl.chat.api.QuiltMessageType;
+
+import java.util.EnumSet;
+
+/**
+ * An abstract message, extended for the various types of messages.
+ * @param <S> The type of the serialized form of this class, usually a Packet<?> but not required
+ */
+public abstract class AbstractChatMessage<S> {
+	protected final @NotNull PlayerEntity player;
+	protected final boolean isOnClientSide;
+
+	protected AbstractChatMessage(@NotNull PlayerEntity player, boolean isOnClientSide) {
+		this.player = player;
+		this.isOnClientSide = isOnClientSide;
+	}
+
+	// These are mostly annotated as @NotNull for dev productivity, I get nice warnings about
+	// What parts of the subclasses I need to implement still
+
+	/**
+	 * @return the set of {@link QuiltMessageType}s associated with this message
+	 */
+	public abstract @NotNull EnumSet<QuiltMessageType> getTypes();
+
+	/**
+	 * @return this message in its serialized form
```

Comment:
```suggestion
	 * {@return this message in its serialized form}
```

Reply:
![image](https://user-images.githubusercontent.com/52360088/212801804-bca70139-cc72-41f1-80d4-696f9099681d.png)
Same thing as client side detection, reader mode both detects it and makes it raw leading to weird duplication

Reply:
Oh no, the duplication is intentional! I just don't know if the `@return` is supposed to be that way or if it was supposed to turn into an english "Return"

Reply:
seems like a very odd choice but alright lol